cmake/modules/header_files.cmake

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Google LLC
+
+include_guard(GLOBAL)
```

Comment:
This file ... feels wrong. @tejlmand

Reply:
cc @nordicjm

Reply:
Keep in mind this is not the final version.
My intention with this is to submit an RFC to test the waters, and see if this approach is acceptable.
I've mentioned that in the comment above, but perhaps I should have been more clear about this.
So yes, it's not particularly pretty. :)

Reply:
At least I need to document it and probably add set it up so that the header is not regenerated every build.

Reply:
not sure why this is made into a CMake file, yet alone a CMake module file.

When introducing new Zephyr CMake modules, then those should be written in a re-usable way, which is surely not the case here.

Reply:
@MaureenHelm we discussed this a bit over a week ago. @kornelduleba and I couldn't figure out a way to make `configure_file` work for what we needed. This approach is a bit clunky but it does seem to work as long as the formatting is maintained. Though I wonder if as a temporary solution it might make sense to just not use C as the interim.

For context, we want to avoid having a separate constant for devicetree and enum. The options being considered are:

1. Have duplication (both the #define and enum) but deprecate the enum for 1 release during which we have to make sure any new channels are added to both
2. Create a common way to generate both from one source.

Personally, due to the complexity we're seeing I'm leaning towards lets just have duplication for 1 release and deprecate the enums.

Reply:
> Personally, due to the complexity we're seeing I'm leaning towards lets just have duplication for 1 release and deprecate the enums.

+1 if it keeps this out of the build system

Reply:
Ok, will do that.